Team Editing/Colored Chips

alanlgirard opened the pull request:
-Added chip colors based on backing type (still need to figure out custom md-chip-templates for the backing selection)
-Added ease of navigation for editing team by author
-Cleaned up Backers/Team layout so it will display properly on smaller screens
-Cleaned up whitespace/newlines on toastSvc.js and accountView.js

@Mctalian @davethenipper @monotkate

Mctalian commented on src/assets/app.css:
```diff
@@ -15,7 +15,7 @@ body {
 
 .md-icon-button {
-    margin-right: 10px;
+  	margin-right: 10px;
```
Is that a tab? Please use 4 spaces instead. I'm surprised jscs didn't catch that one.

Mctalian commented on src/assets/app.css:
```diff
@@ -51,13 +51,57 @@ body {
+  	padding: 3px 10px;
+  	background-color: #e0e0e0;
+  	color: #424242;
+  	border-radius: 10px;
```
Same with all of these lines, looks like somehow there's a mix of spaces and a tab

Mctalian replied:
Like here, if you look at lines 54 through 57 and lines 60 through 64, there's a difference in spacing.

The first goup looks like 4 spaces, the second group looks like 2 spaces and a \t

Mctalian commented on src/assets/app.css:
```diff
@@ -69,19 +113,20 @@ body {
 
 .idea-likes-click {
-    cursor: pointer;
+  	cursor: pointer;
```
Check all of the ones above this line too. Maybe your editor is replacing these things? If you're using Atom check to make sure you're using Soft Tabs

alanlgirard replied:
I have soft tabs enabled.  All of the elements only had 2 spaces before.  I went in and changed them all to 4 spaces so it is standardized throughout all of the files.

davethenipper replied:
Should we do any sort of feedback on the colors?  Maybe this is something
that should be demo'd to our group, or even to the Innovation Challenge
members.

On Tue, Nov 17, 2015 at 2:48 PM, alanlgirard <notifications@github.com>
wrote:

> In src/assets/app.css
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148#discussion_r45110342>
> :
>
> >  }
> >
> >  .idea-likes-click {
> > -    cursor: pointer;
> > +  	cursor: pointer;
>
> I have soft tabs enabled. All of the elements only had 2 spaces before. I
> went in and changed them all to 4 spaces so it is standardized throughout
> all of the files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148/files#r45110342>
> .
>

Mctalian replied:
Ah, ok. And I'm a derp. jscs wouldn't work on a CSS file :facepunch: 

I think it's ok. Some of them look like hard tabs, but it could be GitHub's fault.

It looks good on my end. Approved.

alanlgirard replied:
Some of those were tabs.  Forgot that I changed settings part way through for something else.  Should be all fixed now

alanlgirard replied:
@davethenipper Yes I am open to feedback on the colors.  I just picked random colors off of the materials colors website.

davethenipper replied:
for some reason I'm not seeing this show up in Github... but I'll put this
as an agenda item for the next meeting

On Tue, Nov 17, 2015 at 2:59 PM, alanlgirard <notifications@github.com>
wrote:

> In src/assets/app.css
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148#discussion_r45111960>
> :
>
> >  }
> >
> >  .idea-likes-click {
> > -    cursor: pointer;
> > +  	cursor: pointer;
>
> @davethenipper <https://github.com/davethenipper> Yes I am open to
> feedback on the colors. I just picked random colors off of the materials
> colors website.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148/files#r45111960>
> .
>

davethenipper commented:
Should we do any sort of feedback on the colors?  This might be something we should demo to the Innovation Challenge group.

I need to play around with the colors some to see how they flow with the rest of the site.

alanlgirard commented:
@davethenipper Yes I am open to feedback on the colors. I just picked random colors off of the materials colors website.

Mctalian commented:
The colors remind me of Skittles. :rainbow: 

Also I looked at the Team Stuff as well (I forgot) and it looks like it works pretty well.

I know I already said "Approved" but when there are no teammates, should there by another message similar to the "You're the only Team mate" message?

Mctalian commented:
Oh, also if you add someone other than yourself to the team and remove yourself it still tells you that you're the only one on the team even though you're not.

alanlgirard commented:
Here is a pic of the colors for people that don't want to have to run it.
![back colors](https://cloud.githubusercontent.com/assets/4032278/11224083/30f3bc3c-8d41-11e5-9468-d44eb0f654f5.PNG)

YashdalfTheGray commented:
The colors look good. As the author, should you be able to remove yourself from your own team? Or are we talking about someone other than the author?

Mctalian commented:
We've discussed what if the author comes up with the idea but they don't
really have the time to run with it? That's where the owner backing tag
came in. Ideally that could be passed to someone else on the team. I'm not
sure if that's the same thought that everyone else on the team has, though.
@davethenipper can we perhaps bring this up briefly in the standup after
reports?
On Nov 18, 2015 2:28 AM, "Yash" <notifications@github.com> wrote:

> The colors look good. As the author, should you be able to remove yourself
> from your own team? Or are we talking about someone other than the author?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148#issuecomment-157631181>
> .
>

bcbrennecke commented:
Is that the card I'm working on?  The owner suggesting what type of backing they need

Sent from my iPhone

> On Nov 18, 2015, at 8:06 AM, Mctalian <notifications@github.com> wrote:
> 
> We've discussed what if the author comes up with the idea but they don't
> really have the time to run with it? That's where the owner backing tag
> came in. Ideally that could be passed to someone else on the team. I'm not
> sure if that's the same thought that everyone else on the team has, though.
> @davethenipper can we perhaps bring this up briefly in the standup after
> reports?
> On Nov 18, 2015 2:28 AM, "Yash" <notifications@github.com> wrote:
> 
> > The colors look good. As the author, should you be able to remove yourself
> > from your own team? Or are we talking about someone other than the author?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/YashdalfTheGray/flintandsteel/pull/148#issuecomment-157631181>
> > .
> >
> —
> Reply to this email directly or view it on GitHub.
>

davethenipper commented:
Sure, I'll add it to the agenda.  And that is not the card you are working on Braun, so no worries.

Also, approved.

alanlgirard commented:
Added a function to check if a user is a member of the team and cleaned up the text that was being displayed to properly reflect the state.  Everything checks out except for the instance where you navigate to the teams tab while logged out and then log in.  The info doesn't update until a navigation to another tab and back.  I will log an issue so this pull request can get through.

@Mctalian @davethenipper can you re-approve?

YashdalfTheGray commented:
Is there some place that you can call the refresh idea button? I forget if
there is one or not but I think @Mctalian made one at some point.

Yash Kulshrestha
yash.kulshrestha@gmail.com
(614)563-1163
Sent from my Nexus 6P
On Nov 18, 2015 12:02 PM, "alanlgirard" <notifications@github.com> wrote:

> Added a function to check if a user is a member of the team and cleaned up
> the text that was being displayed to properly reflect the state. Everything
> checks out except for the instance where you navigate to the teams tab
> while logged out and then log in. The info doesn't update until a
> navigation to another tab and back. I will log an issue so this pull
> request can get through.
>
> @Mctalian <https://github.com/Mctalian> @davethenipper
> <https://github.com/davethenipper> can you re-approve?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YashdalfTheGray/flintandsteel/pull/148#issuecomment-157803619>
> .
>

davethenipper commented:
Approved!

Mctalian commented on src/ideas/ideasView/ideasView.js:
```diff
@@ -285,6 +295,15 @@ angular.module('flintAndSteel')
+                ctrl.isInTeam = false;
+                for (var i = 0; i < $scope.idea.team.length; i++) {
+                    if (loginSvc.isUserLoggedIn() && loginSvc.getProperty('_id') === $scope.idea.team[i].memberId) {
+                        ctrl.isInTeam = true;
```
I recommend a `break;` after this line.

Mctalian replied:
On second thought, let's restructure this a bit. Instead of setting a ctrl.isInTeam flag, we can `return true;` in the loop and `return false;` outside of the loop.

Then you can remove this call from `md-on-select` in the HTML, and replace your `ideasView.isInTeam` with this call instead. I tested it locally it works like a charm.

Mctalian replied:
This apparently causes a lot of console errors to appear, though. We'll have to look into this later.

Mctalian commented on src/ideas/ideasView/ideasView.tpl.html:
```diff
@@ -142,54 +145,88 @@
-                                    <div class="md-list-item-text">
-                                        No Team has been formed yet :)
+                                        <span class="custom-chip-wrapper" ng-repeat="type in back.types">
+                                            <small class="custom-chip {{type.name}}">{{type.name}}</small>
```
For the class, the backing types do have `_lowername` properties that currently have no use.

If you got to IdeaSvc.js line ~100, replace the existing type._lowername with this:

`type._lowername = type.name.toLowerCase().replace(/[ ]/g, '-').replace('?', '');`

That takes the name, makes it lowercase, uses regular expressions to find all of the spaces and convert them to dashes, then removes any question marks. Giving you perfect CSS class names, if you want them. Works fine the way it is now.

alanlgirard replied:
That method is a lot cleaner and works like a charm!  I always like having perfect css classes.  My first method was kind of a hack way of getting it to work.

Mctalian commented:
At the very least, add the `break;` in the line comments above. Everything else is up to you.

Approved!

Mctalian commented:
In other words... /play pushit

Mctalian commented:
Class changes look good and all of the scenarios I tried worked, but we DO get console errors unfortunately. At least on my end. We will look at this tonight and merge later.

Mctalian commented:
Approved by team.
